### PR TITLE
ADAPT-2918: Story Card

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Story Card (vertical) with minimal variant
- Options for: big text (title and teaser), minimal, hide image (will also hide tab automatically), hide tab only, heading level, image focus
- Note: Waiting for Story Card media icon (for podcast/video) in mockup - will add this in a later PR.
- Note2: no need to review the storyPageView.js code for now - work will be continued in another PR for story page.

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. https://deploy-preview-63--stanford-alumni.netlify.app/test/yvonne/test-story-cards/
2. Look at all the options
3. Pull down locally and go to Text Event page on Storyblok http://app.storyblok.com/#!/me/spaces/103430/stories/0/0/49870707
4. Play around with the story card options
5. Look at code

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: src/components/content-types/event/event.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ const Event = ({
   }
 
   let wrapperClasses =
-    "su-rs-pb-3 su-bg-white su-border su-border-solid su-border-black-10 su-shadow-sm";
+    "su-rs-pb-3 su-bg-white su-border su-border-solid su-border-transparent-black su-bg-clip-padding su-shadow-sm";
```
Comment (yvonnetangsu) [inline src/components/content-types/event/event.js:76]: Using `background-clip: padding box` with combination of translucent border makes it look good on both light and dark section backgrounds.

Comment (sherakama) [reply src/components/content-types/event/event.js:76]: neat.

File: src/tailwind/plugins/theme/colors.js [hunk 2]
```diff
@@ -4,7 +4,7 @@
 module.exports = function () {
   return {
     "transparent-black": {
-      DEFAULT: "rgba(0, 0, 0, 10%)", // For card borders
+      DEFAULT: "rgba(200, 200, 200, 20%)", // For card borders
```
Comment (sherakama) [inline src/tailwind/plugins/theme/colors.js:7]: This is not black anymore. Perhaps a name change is needed.

File: src/components/layout/grid.js [hunk 3]
```diff
@@ -35,7 +36,7 @@ const Grid = ({
         itemClasses
       )}
     >
-      <CreateBloks blokSection={content} />
+      <CreateBloks blokSection={content} isDark={isDark} />
```
Comment (sherakama) [inline src/components/layout/grid.js:39]:
```suggestion
      <CreateBloks blokSection={content} isDark />
```

File: src/components/layout/grid.js [hunk 4]
```diff
@@ -48,7 +49,7 @@ const Grid = ({
           itemClasses
         )}
       >
-        <CreateBloks blokSection={content} />
+        <CreateBloks blokSection={content} isDark={isDark} />
```
Comment (sherakama) [inline src/components/layout/grid.js:52]:
```suggestion
        <CreateBloks blokSection={content} isDark />
```

File: src/components/cards/storyCard.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    hideImage,
+    headingLevel,
+    cardImageFocus,
+  },
+  isDark,
+}) => (
+  <CreateStories
+    stories={storyPicker}
+    layout="card"
+    isBigText={isBigText}
```
Comment (sherakama) [inline src/components/cards/storyCard.js:19]: Usually, the linter complains to me about the short syntax here: 

```suggestion
    isBigText
```

File: src/components/content-types/perk/perkPageView.js [hunk 6]
```diff
@@ -7,14 +7,7 @@ import Layout from "../../partials/layout";
 const PerkPageView = (props) => {
   // Destructure props
   const {
-    blok: {
-      image: { filename } = {},
-      imageFocus,
-      isNew,
-      title,
-      intro,
-      externalUrl,
```
Comment (sherakama) [inline src/components/content-types/perk/perkPageView.js:16]: Why is `externalUrl` being removed here?

File: src/components/cards/storyCard.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+const StoryCard = ({
+  blok: {
+    storyPicker,
+    isMinimal,
+    isBigText,
+    hideTab,
+    hideImage,
+    headingLevel,
+    cardImageFocus,
```
Comment (sherakama) [inline src/components/cards/storyCard.js:12]: Where does this come from? I am not seeing it in storyblok schema?

File: src/components/page-types/darkPage.js [hunk 8]
```diff
@@ -17,13 +17,6 @@ const DarkPage = (props) => {
   const numContent = getNumBloks(content);
   const numBelow = getNumBloks(belowContent);
 
-  // Only add top padding to Main Content if the Above Content region is populated or if page title is visually hidden
-  let contentPadding = "su-rs-pb-7";
-
-  if (numAbove > 0 || isSrOnlyTitle) {
-    contentPadding = "su-rs-py-7";
-  }
-
```
Comment (sherakama) [inline src/components/page-types/darkPage.js:26]: This seems unrelated. Are you sure you want these changes?

Comment (yvonnetangsu) [reply src/components/page-types/darkPage.js:26]: Yeah I kind of sneaked this change in here. This part is irrelevant for dark pages for now.

File: src/components/cards/storyCard.js (hunk 7 above)
Comment (yvonnetangsu) [reply src/components/cards/storyCard.js:12]: These are in the individual story picker (Story Card component) in Storyblok and not in the Story content type schema. The idea is that the story fields can be the same, but the card version can be placed in multiple places with different options enabled.

File: src/components/content-types/perk/perkPageView.js (hunk 6 above)
Comment (yvonnetangsu) [reply src/components/content-types/perk/perkPageView.js:16]: This is another clean up that I sneaked in for perk page. We don't have a perk page for MVP, but the structured is set up. However, the external URL will not be needed if an internal perk page exists so it can be removed.

**Comment by sherakama** [comment]
I added an anchor icon to the card view information but I don't see it anywhere.

**Comment by yvonnetangsu** [comment]
> I added an anchor icon to the card view information but I don't see it anywhere.

Oh, the icon field is still being designed. So will add in a later PR (updated the summary to reflect this).

File: src/components/cards/storyCard.js (hunk 7 above)
Comment (sherakama) [reply src/components/cards/storyCard.js:12]: Yeah, sorry, I was looking at the story type and not the story card schema.

File: src/components/layout/grid.js (hunk 4 above)
Comment (yvonnetangsu) [reply src/components/layout/grid.js:52]: Committed the one above which I think is the same

File: src/components/content-types/story/storyCardView.js [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+    title,
+    shortTitle,
+    teaser,
+    intro,
+    source,
+    pubLink,
+    tabText,
+  },
+  blok,
+  storyLink,
```
Comment (sherakama) [inline src/components/content-types/story/storyCardView.js:24]: Where does this come from?

File: src/components/content-types/story/storyCardView.js [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+          <SbLink
+            link={pubLink || storyLink}
+            classes={dcnb(
+              "su-stretched-link su-z-20 su-rs-mt-2 su-mb-0 su-no-underline hocus:su-underline su-underline-offset !su-underline-thick !su-underline-digital-red-xlight",
+              headlineSize,
+              headlineColor
+            )}
+          >
+            <Heading
+              level={headingLevel ?? 3}
```
Comment (sherakama) [inline src/components/content-types/story/storyCardView.js:116]:
```suggestion
              level={parseInt(headingLevel, 10) ?? 3}
```

File: src/components/content-types/story/storyPageView.js [hunk 11]
```diff
@@ -0,0 +1,49 @@
+import SbEditable from "storyblok-react";
+import React from "react";
+import { Heading } from "decanter-react";
+import CardImage from "../../media/cardImage";
+import Layout from "../../partials/layout";
+import CreateBloks from "../../../utilities/createBloks";
+
+const StoryPageView = (props) => {
```
Comment (sherakama) [inline src/components/content-types/story/storyPageView.js:8]: Not reviewing this file as you will work on it later.

**Review by sherakama — CHANGES_REQUESTED** [review]
Couple of smaller changes and then this is GTG.

File: src/components/content-types/story/storyCardView.js (hunk 9 above)
Comment (yvonnetangsu) [reply src/components/content-types/story/storyCardView.js:24]: This is passed down from CreateBloks - it grabs the full slug associated with that blok

File: src/components/content-types/story/storyCardView.js (hunk 10 above)
Comment (yvonnetangsu) [reply src/components/content-types/story/storyCardView.js:116]: Duh - forgot this one. Thank you!

File: src/components/content-types/story/storyCardView.js (hunk 9 above)
Comment (sherakama) [reply src/components/content-types/story/storyCardView.js:24]: TY

File: src/components/layout/grid.js (hunk 4 above)
Comment (yvonnetangsu) [reply src/components/layout/grid.js]: Actually I have to revert these since this is setting it as isDark always true instead of passing the boolean option down.

File: src/tailwind/plugins/theme/colors.js [hunk 12]
```diff
@@ -3,9 +3,17 @@
...
+      true: {
+        transparent: {
+          20: "rgba(0, 0, 0, 20%)", // For use in linear gradients
+        },
+      },
+      30: {
+        transparent: {
+          20: "rgba(192, 192, 191, 20%)", // For card borders
+        },
+      },
```
Comment (yvonnetangsu) [inline src/tailwind/plugins/theme/colors.js:16]: @sherakama I think this should be more future-proof now. So for example, su-border-black-30-transparent-20 would mean use the color "30% black" at its 20% opacity. Actually, let me make one more change - I like opacity better than transparent.

**Review by sherakama — APPROVED** [review]
OK, GTG.
